crates/context_aware_config/src/api/context/handlers.rs

```diff
@@ -575,32 +592,59 @@ async fn get_context(
 
 #[get("/list")]
 async fn list_contexts(
-    qparams: Query<PaginationParams>,
+    pagination_params: PlatformQuery<PaginationParams>,
+    dimension_params: DynamicQuery<QueryMap>,
     db_conn: DbConnection,
-) -> superposition::Result<impl Responder> {
+) -> superposition::Result<Json<Vec<Context>>> {
     use crate::db::schema::contexts::dsl::*;
     let DbConnection(mut conn) = db_conn;
 
-    let PaginationParams {
-        page: opt_page,
-        size: opt_size,
-    } = qparams.into_inner();
-    let default_page = 1;
-    let page = opt_page.unwrap_or(default_page);
-    let default_size = 20;
-    let size = opt_size.unwrap_or(default_size);
+    let pagination_params = pagination_params.into_inner();
+    let page = pagination_params.page.unwrap_or(1);
+    let size = pagination_params.size.unwrap_or(10);
 
     if page < 1 {
         return Err(bad_argument!("Param 'page' has to be at least 1."));
     } else if size < 1 {
         return Err(bad_argument!("Param 'size' has to be at least 1."));
     }
 
-    let result: Vec<Context> = contexts
-        .order(created_at)
-        .limit(i64::from(size))
-        .offset(i64::from(size * (page - 1)))
-        .load(&mut conn)?;
+    let dimension_params = dimension_params.into_inner().into_inner();
+
+    let result = if dimension_params.len() > 0 || pagination_params.prefix.is_some() {
+        let mut all_contexts: Vec<Context> =
+            contexts.order(created_at).load(&mut conn)?;
+        if let Some(prefix) = pagination_params.prefix {
+            let prefix_list = prefix.split(',').map(String::from).collect::<HashSet<_>>();
+            all_contexts = all_contexts
+                .into_iter()
+                .filter_map(|mut context| {
+                    Context::filter_keys_by_prefix(&context, &prefix_list)
+                        .map(|filtered_overrides_map| {
+                            context.override_ = filtered_overrides_map.into_inner();
+                            context
+                        })
+                        .ok()
+                })
+                .collect()
+        }
+        let dimension_keys = dimension_params.keys().cloned().collect::<Vec<_>>();
+        let dimension_filter_contexts =
+            Context::filter_by_dimension(all_contexts, &dimension_keys);
+        let eval_filter_contexts =
+            Context::filter_by_eval(dimension_filter_contexts, &dimension_params);
+        let start = (size * (page - 1)) as usize;
+        let end = min((size * page) as usize, eval_filter_contexts.len());
+        eval_filter_contexts
+            .get(start..end)
+            .map_or(vec![], |slice| slice.to_vec())
+    } else {
+        contexts
+            .order(created_at)
+            .limit(i64::from(size))
+            .offset(i64::from(size * (page - 1)))
+            .load::<Context>(&mut conn)?
+    };
```

Comment:
INFO: changes for dimension based and prefix based filtering in `/context/list` endpoint